src/common/function/src/scalars/geo/geohash.rs

```diff
@@ -130,6 +132,144 @@ impl Function for GeohashFunction {
 
 impl fmt::Display for GeohashFunction {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{}", NAME)
+        write!(f, "{}", Self::NAME)
+    }
+}
+
+/// Function that return geohash string for a given geospatial coordinate.
+#[derive(Clone, Debug, Default)]
+pub struct GeohashNeighboursFunction;
+
+impl GeohashNeighboursFunction {
+    const NAME: &'static str = "geohash_neighbours";
+}
+
+impl Function for GeohashNeighboursFunction {
+    fn name(&self) -> &str {
+        GeohashNeighboursFunction::NAME
+    }
+
+    fn return_type(&self, _input_types: &[ConcreteDataType]) -> Result<ConcreteDataType> {
+        Ok(ConcreteDataType::list_datatype(
+            ConcreteDataType::string_datatype(),
+        ))
+    }
+
+    fn signature(&self) -> Signature {
+        let mut signatures = Vec::new();
```

Comment:
You can use 
```rust
helper::one_of_sigs2(vec![ 
            ConcreteDataType::float32_datatype(),
            ConcreteDataType::float64_datatype()],
            [
               ConcreteDataType::int8_datatype(),
                ConcreteDataType::int16_datatype(),
                ConcreteDataType::int32_datatype(),
                ConcreteDataType::int64_datatype(),
                ConcreteDataType::uint8_datatype(),
                ConcreteDataType::uint16_datatype(),
                ConcreteDataType::uint32_datatype(),
                ConcreteDataType::uint64_datatype(),
             ])
```
instead.

https://github.com/GreptimeTeam/greptimedb/blob/36b1bafbf00152cc89c8d4362ace432bbb1932a3/src/common/function/src/helper.rs#L23

Reply:
the helper only works with 2-argument signature. In this case there is 3, especially the frist two are in the same type. That's a little bit different from general use-case